via/via_system_macos.cpp

```diff
@@ -0,0 +1,1435 @@
+/*
+ * Copyright (c) 2016-2019 Valve Corporation
+ * Copyright (c) 2016-2019 LunarG, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Mark Young <marky@lunarg.com>
+ * Author: Jeremy Kniager <jeremyk@lunarg.com>
+ */
+
+#ifdef VIA_MACOS_TARGET
+
+#include <cstring>
+#include <sstream>
+#include <algorithm>
+
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/statvfs.h>
+#include <sys/utsname.h>
+#include <dirent.h>
+#include <unistd.h>
+#include <dlfcn.h>
+
+#include "via_system_macos.hpp"
+
+ViaSystemMacOS::ViaSystemMacOS() : ViaSystem() {
+    char temp_c_string[1024];
+    ssize_t len = ::readlink("/proc/self/exe", temp_c_string, 1023);
```

Comment:
I think you may want to use _NSGetExecutablePath() here to read the current process' path; I don't think /proc is available on MacOS.